lib/shoulda/matchers/action_controller/callback_matcher.rb

```diff
@@ -157,27 +278,39 @@ def initialize(method_name, kind, callback_type)
           @method_name = method_name
           @kind = kind
           @callback_type = callback_type
+          @options = {}
+        end
+
+        def only(actions)
+          @options[:only] = Array(actions)
+          self
         end
 
         def matches?(controller)
           @controller = controller
           @controller_class = controller.class
 
-          callbacks.map(&:filter).include?(method_name)
+          filter_matches? && only_matches?
         end
 
         def failure_message
-          "Expected that #{controller_class.name} would have :#{method_name} as a #{kind}_#{callback_type}"
+          message = "Expected that #{controller_class.name} would have :#{method_name} as a #{kind}_#{callback_type}"
```

Comment:
Instead of extracting a method that modifies a string, what do you think about rephrasing this failure message so you can further simplify `failure_message` and `failure_message_when_negated`? In other matchers we've started adopting this pattern:

``` ruby
def description
  expectation
end

def failure_message
  "Expected #{controller_class.name} to #{expectation}"
end

def failure_message_when_negated
  "Expected #{controller_class.name} not to #{expectation}"
end

private

def expectation
  # here you can also include any `@options` which have been set
  "have :#{method_name} as a #{kind}_#{callback} ......."
end
```